File: java-code/src/main/java/refactoring/basics/extract_variable/ExtractVariable_After.java
package refactoring.basics.extract_variable;

import refactoring.basics.model.Order;

public class ExtractVariable_After {

  /**
   * This method can actually be moved to order class as a method as all the
   * variables are from order class.
   */
  public double price(Order order) {
    final double basePrice = order.getQuantity() * order.getItemPrice();
    final double quantityDiscount = Math.max(0, order.getQuantity() - 500) * 0.05;
    final double shipping = Math.min(basePrice * 0.1, 100);

    return basePrice - quantityDiscount + shipping;
  }
}

File: go-code/basics/extract_variable/extract-variable_after.go
package extract_variable

import (
	"gitlab.com/clean-code55/refactor/improving-design-of-existing-code/go-code/basics/models"
	"math"
)

// this func can be moved to Order
func priceAfter(order models.Order) float64 {
	// price = base price - quantity discount + shipping
	basePrice := float64(order.Quantity) * order.ItemPrice
	orderDiscount := math.Max(0, float64(order.Quantity-500)) * 0.05
	shipping := math.Min(basePrice*0.1, 100)

	return basePrice - orderDiscount + shipping
}
